### PR TITLE
Refactor labchecker to openlabcmd

### PR DESCRIPTION
**Pull request by wangxiyuan**
Refactor labchecker and provide a common command line
tool for openlab management.

The command `python labchecker.py ...` has been changed to
`openlab check ...`

More command will be added in the future like `openlab cluster ...`
for openlab HA management.

### CODE REVIEW

**Review by Yikun — CHANGES_REQUESTED** [review]
The plugin import also need to be changed
"from labcheck import Plugin" --> "from openlabclient import Plugin"

**Comment by liusheng** [comment]
openlab is not a generic software or component... so openlabclient looks like a bit werid :)

**Comment by wangxiyuan** [comment]
> openlab is not a generic software or component... so openlabclient looks like a bit werid :)

This is just used internally for environment management. The name can be whatever. Any suggestion?

**Comment by liusheng** [comment]
> > openlab is not a generic software or component... so openlabclient looks like a bit werid :)
> 
> This is just used internally for environment management. The name can be whatever. Any suggestion?

how about just use the `labchecker` as command prefix ? e.g. `labchecker securitygroup`... as the `xxxclient` is counter to a "service side", but `openlab` is not a service side.

**Comment by Yikun** [comment]
> openlab is not a generic software or component... so openlabclient looks like a bit werid :)

IMO, `openlab` is lgtm, and we can use the command like below to do some check.
```
openlab check nodepool
```

and for future use, we can also use:
```
openlab ha/cluster xxx
```

That means we do some action in openlab, and openlabclient just like the collection of these tools.

File: openlabclient/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+This is the command line tool for OpenLab management.
+
+## How to use
+
+You can install from source or just use `pip install openlabclient`
+
+## Supported command
+
+### Labchecker
```
Comment (Yikun) [inline openlabclient/README.md]: Better to say:
## Supported feature
``openlab check``: Health check tool for checking OpenLab CI infrastructures.

File: openlabclient/openlabclient/cli.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    def create_parser(self):
+        parser = argparse.ArgumentParser(
+            description='The command line tool for OpenLab management',
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+        parser.add_argument('-c', dest='config',
+                            default='/etc/openlab/openlab.yaml',
+                            help='path to config file')
+
+        subparsers = parser.add_subparsers(title='commands',
```
Comment (Yikun) [inline openlabclient/openlabclient/cli.py]:
```
# openlab -h
usage: openlab [-h] [-c CONFIG] {check} ...

The command line tool for OpenLab management

optional arguments:
  -h, --help  show this help message and exit
  -c CONFIG   path to config file (default: /etc/openlab/openlab.yaml)

commands:
  valid commands                                             <====  redundant

  {check}     additional help                               <====  redundant
    check     check openlab environment
```

File: openlabclient/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+Health check tool for checking OpenLab CI infrastructures.
```
Comment (kiwik) [inline openlabclient/README.md]: Please give a command example at here.

File: openlabclient/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# OpenLab management tool
+
+This is the command line tool for OpenLab management.
+
+## How to use
+
+You can install from source or just use `pip install openlabclient`
```
Comment (Yikun) [inline openlabclient/README.md]: Really? Currently, we can only use:
```
git clone https://github.com/theopenlab/labkeeper.git
cd labkeeper/openlabclient/
python setup.py install
```

File: openlabclient/openlabclient/cli.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    def __init__(self):
+        self.parser = None
+        self.args = None
+
+    def create_parser(self):
+        parser = argparse.ArgumentParser(
+            description='The command line tool for OpenLab management',
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+        parser.add_argument('-c', dest='config',
```
Comment (Yikun) [inline openlabclient/openlabclient/cli.py]: TODO: Not implemented yet.

File: openlabclient/openlabclient/cli.py [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                                    "'vexxhost'. Default is 'all'.")
+        cmd_check.add_argument('--nocolor', action='store_true',
+                               help='Enable the no color mode.')
+        cmd_check.add_argument('--recover', action='store_true',
+                               help='Enable the auto recover mode.')
+
+        return parser
+
+    def _get_cloud_list(self, cloud):
+        with open('/etc/openstack/clouds.yaml') as f:
```
Comment (kiwik) [inline openlabclient/openlabclient/cli.py]: I see a `-c` option in line#20, this is hard code?

**Review by Yikun — CHANGES_REQUESTED** [review]

File: openlabclient/openlabclient/cli.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+            clouds = yaml.load(f, Loader=yaml.FullLoader)
+            clouds_list = [c for c in clouds['clouds']]
+
+        if cloud not in clouds_list + ['all']:
+            print("Error: Cloud %s is not found." % cloud)
+            print("Please use the cloud in %s." % clouds_list)
+            exit(2)
```
Comment (kiwik) [inline openlabclient/openlabclient/cli.py]: `sys.exit(main())` in line#109, this is other exit point?

File: openlabclient/README.md (hunk 4 above)
Comment (Yikun) [reply openlabclient/README.md]:
```
# pip search openlabclient
  INSTALLED: 0.0.1.dev642
  LATEST:    0.0.1.dev643
```

Cool, recycle my guess.

File: openlabclient/README.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# OpenLab management tool
```
Comment (huangtianhua) [inline openlabclient/README.md]: How about name this openlabtool instead of openlabclient?

File: openlabclient/README.md (hunk 1 above)
Comment (huangtianhua) [reply openlabclient/README.md]: +1

File: openlabclient/openlabclient/cli.py (hunk 6 above)
Comment (Yikun) [reply openlabclient/openlabclient/cli.py]: This cloud config is different from the openlab config, In future, we can use below command to specify the openlab config:
```
openlab check --config /etc/openlab/openlab.yaml
```
and this ``'/etc/openstack/clouds.yaml'`` can be set in the conif like
```
# cat  /etc/openlab/openlab.yaml

[default]
cloud_yaml = /etc/openstack/clouds.yaml
```

File: openlabclient/openlabclient/cli.py [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        return parser
+
+    def _get_cloud_list(self, cloud):
+        with open('/etc/openstack/clouds.yaml') as f:
+            clouds = yaml.load(f, Loader=yaml.FullLoader)
+            clouds_list = [c for c in clouds['clouds']]
+
+        if cloud not in clouds_list + ['all']:
+            print("Error: Cloud %s is not found." % cloud)
+            print("Please use the cloud in %s." % clouds_list)
```
Comment (huangtianhua) [inline openlabclient/openlabclient/cli.py]: print("Please use the cloud in %s." % clouds_list or using 'all').

**Comment by liusheng** [comment]
> > openlab is not a generic software or component... so openlabclient looks like a bit werid :)
> 
> IMO, `openlab` is lgtm, and we can use the command like below to do some check.
> 
> ```
> openlab check nodepool
> ```
> and for future use, we can also use:
> 
> ```
> openlab ha/cluster xxx
> ```
> That means we do some action in openlab, and openlabclient just like the collection of these tools.

maybe `openlab` command is ok, but the `openlabclient` looks like a bit werid, there is not something like `openlabserver`, just personal opinion.

File: openlabclient/README.md (hunk 1 above)
Comment (wangxiyuan) [reply openlabclient/README.md]: Done

File: openlabclient/openlabclient/cli.py (hunk 5 above)
Comment (wangxiyuan) [reply openlabclient/openlabclient/cli.py]: Done

File: openlabclient/openlabclient/cli.py (hunk 6 above)
Comment (wangxiyuan) [reply openlabclient/openlabclient/cli.py]: ++ for @Yikun .While this pr is only for interface refactor.  `-c` doesn't work at this moment. This is just a placeholder. We can cooperate to update the tools after this refactor.

File: openlabclient/openlabclient/cli.py (hunk 7 above)
Comment (wangxiyuan) [reply openlabclient/openlabclient/cli.py]: Done

File: openlabclient/README.md (hunk 3 above)
Comment (wangxiyuan) [reply openlabclient/README.md]: Done

File: openlabclient/openlabclient/cli.py (hunk 2 above)
Comment (wangxiyuan) [reply openlabclient/openlabclient/cli.py]: Done

**Comment by wangxiyuan** [comment]
@h00130372 @liu-sheng  I changed the name to openlabcmd. works for you?

File: openlabclient/openlabclient/cli.py (hunk 9 above)
Comment (wangxiyuan) [reply openlabclient/openlabclient/cli.py]: Done

File: openlabclient/openlabclient/cli.py (hunk 6 above)
Comment (kiwik) [reply openlabclient/openlabclient/cli.py]: It's OK to me.

**Review by kiwik — APPROVED** [review]
LGTM

**Review by liusheng — APPROVED** [review]
lgtm

**Pull request merged**